### ISSUE TITLE
[Bar chart] "Color by" feature

### ISSUE THREAD

**Issue by ValentinC-BR** [apache/superset#14021]
## Screenshot

![image](https://user-images.githubusercontent.com/79460908/114021786-5cc3e300-9871-11eb-990c-9855416b713d.png)

## Description

As the color/legend is based on the breakdown and the breakdown cannot be the same as the series column, there is no way to have bars with various color (one per bar) in  **Bar Charts**.

The possibility to color by 'series' instead of 'breakdown' would be very appreciated.

Use case : I want one bar per customer, and the same colors as in the rest of the dashboard, to easily see which one is the most important, and where is my customer on each chart, without reading the legend each time.

## Design input

/

**Comment by junlincc** [comment]
@ValentinC-BR 
I am curious, how would adding a color layer to a simple bar chart help user see the difference of the bar? wouldn't it be a distraction instead? 

@mihir174 @steejay wdyt?

**Comment by ValentinC-BR** [comment]
As pie charts are pretty misleading, I'd rather replace them with bar charts.

However, contrary to pie chats (in Superset), I can't easily identify (i mean without reading) where is the customer A in each chart)
If customer A is always the blue bar, I find it easier to do so.

I might be wrong however.

**Comment by rusackas** [comment]
While this is an interesting feature, I don't think it's on at least our roadmap for now. We can consider it a little further when we fully replace the categorical bar charts in the future. In the meantime, would you be interested in opening a PR to tackle this?

@villebro since it may pertain to something on your eventual roadmap.

**Comment by ValentinC-BR** [comment]
Hi,
I'm actually an end user (data analyst), and I don't feel like I'll be able to make a "good" PR for this. I'd rather let others make it for now.
Thanks for proposing.

**Comment by Steejay** [comment]
> I am curious, how would adding a color layer to a simple bar chart help user see the difference of the bar? wouldn't it be a distraction instead?

I don't think it would be a distraction. The solution/feature suggestion makes sense. Unique colors would make it easier to identify the different customers in the use case presented.
